report/renderer/details-renderer.js

```diff
@@ -376,7 +379,106 @@ export class DetailsRenderer {
+   * @param {HTMLTableRowElement} rowEl
+   */
+  _adornTableRowWithEntityChips(rowEl) {
+    const entityName = rowEl.dataset.entity;
+    if (!entityName) return;
+    const matchedEntity = this._entities?.find(e => e.name === entityName);
+    if (!matchedEntity) return;
+
+    const firstTdEl = this._dom.find('td', rowEl);
+
+    if (matchedEntity.category) {
+      const categoryChipEl = this._dom.createElement('span');
+      categoryChipEl.classList.add('lh-audit__adorn');
+      categoryChipEl.textContent = matchedEntity.category;
+      firstTdEl.append(' ', categoryChipEl);
+    }
+
+    if (matchedEntity.isFirstParty) {
+      const firstPartyChipEl = this._dom.createElement('span');
+      firstPartyChipEl.classList.add('lh-audit__adorn', 'lh-audit__adorn1p');
+      firstPartyChipEl.textContent = Globals.strings.firstPartyChipLabel;
+      firstTdEl.append(' ', firstPartyChipEl);
+    }
+
+    if (matchedEntity.homepage) {
+      const entityLinkEl = this._dom.createElement('a');
+      entityLinkEl.href = matchedEntity.homepage;
+      entityLinkEl.target = '_blank';
+      entityLinkEl.title = Globals.strings.openInANewTabTooltip;
+      entityLinkEl.classList.add('lh-report-icon--external');
+      firstTdEl.append(' ', entityLinkEl);
+    }
+  }
+
+  /**
+   * Renders an aggregated row.
+   * @param {TableItem} item
+   * @param {LH.Audit.Details.TableColumnHeading[]} headings
+   */
+  _renderEntityGroupRow(item, headings) {
+    const entityColumnHeading = {...headings[0]};
+    // In subitem-situations (unused-js), ensure Entity name is not rendered as code, etc.
+    entityColumnHeading.valueType = 'text';
+    const aggregateRowHeadings = [entityColumnHeading, ...headings.slice(1)];
+    const fragment = this._dom.createFragment();
+    fragment.append(this._renderTableRow(item, aggregateRowHeadings));
+    return fragment;
+  }
+
+  /**
+   * Computes entity group items and column totals from TableItem's
+   * @param {TableLike} details
+   * @return {TableItem[]}
+   */
+  _getEntityGroupItems(details) {
+    const {items, headings, sortedBy} = details;
+    // Exclude pre-aggregated audits and results without entity classification.
+    // Eg. Third-party Summary comes pre-aggregated.
+    if (!items.length || details.isEntityGrouped || !items.some(item => item.entity)) {
+      return [];
+    }
+    const skippedColumns = new Set(details.skipSumming || []);
+    const supportedAggregations = ['bytes', 'numeric', 'ms', 'timespanMs'];
+    /** @type {string[]} */
+    const aggregateKeys = [];
+    for (const heading of headings) {
+      if (!heading.key || skippedColumns.has(heading.key)) continue;
+      if ('valueType' in heading && supportedAggregations.includes(heading.valueType)) {
+        aggregateKeys.push(heading.key);
+      }
+    }
+
+    // Grab the first column's key to aggregate by entity
+    const primaryKey = headings[0].key;
+    if (!primaryKey) return [];
```

Comment:
I'm not clear on the function of primary key, can you elaborate? Does this enforce the first heading of an audit to meet a specific constraint to be grouped like this?

Reply:
Could things become simpler if the Table/Opp details had a `aggregationHeading?: {keys: string[], sortBy?: string[], displayUnit?: string, granularity?: number}`? This would match the existing `subItemsHeading`.

Reply:
`aggregationHeading?: {keys: string[], sortBy?: string[]}` sounds good unless we plan to use the `sortBy` information elsewhere besides 3p stuff. I was imagining a future where `sortBy` passed to `makeTableDetails` will actively sort the items without manually doing it elsewhere in the audit. In that case it would make sense to keep them separate.

That was just an idea though, there are clearly multiple ways to do this. I'll defer to @alexnj on this one.

Reply:
> [re: hypothetical use case] without manually doing it elsewhere in the audit

It's gotta be declared in the audit anyway. Either with a `sortBy` property, or a call to something like `Audit.makeTableDetail(..., {sortBy: []})`. I don't see why the former would be preferred over the latter.

Reply:
I expect more of the aggregation to move to audit time, over time. I like the idea of `makeTableDetails` doing the sorting (at least for the simple cases, which is the majority). 

What would `keys: string[]` contain? The format isn't really going match the existing heading/subHeading structure, so I see it as just moving from one structure to another. `aggregationHeading: false` doesnt necessary translate to `isEntityGrouped`, one indicates a directive to group items, and the latter indicates that the audit result already comes grouped.